decouple.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+import collections
+import fnmatch
+import json
+import pathlib
+import subprocess
+import sys
+
+import yaml
+
+CURRENT_DIR = pathlib.Path(__file__).parent.absolute()
+
+
+def generate_dependency_graph(*args):
+    argstring = ' '.join(args)
+    command = f'pydeps --nodot --show-deps {argstring}'
+    print(f"Running: {command}")  # noqa: T201
+    result = subprocess.run(
+        command, shell=True, check=True, capture_output=True, text=True
```

Comment:
Is `shell=True` necessary? It should only be needed if you want to expand globs or propagate environment variables, otherwise it's best avoided.

See [Python security considerations for `shell=True`](https://docs.python.org/3/library/subprocess.html#security-considerations) and [this related SO answer](https://stackoverflow.com/a/3172488) for the rationale.

Reply:
Not at all, removing.